File: lib/features/rooms/domain/entities/room_member.dart
/// Room member entity
class RoomMember {
  final String id;
  final String roomId;
  final String userId;
  final String role; // 'admin' or 'member'
  final DateTime joinedAt;

  /// Populated from join with profiles
  final String? username;
  final String? displayName;
  final String? avatarUrl;

  const RoomMember({
    required this.id,
    required this.roomId,
    required this.userId,
    required this.role,
    required this.joinedAt,
    this.username,
    this.displayName,
    this.avatarUrl,
  });

  bool get isAdmin => role == 'admin';

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is RoomMember && other.id == id;
  }

  @override
  int get hashCode => id.hashCode;

  @override
  String toString() {
    return 'RoomMember(userId: $userId, role: $role, username: $username)';
  }
}

File: backend/src/main/java/com/kanzi/api/room/RoomMember.java
package com.kanzi.api.room;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "room_members")
@Getter
@Setter
@NoArgsConstructor
public class RoomMember {

    public static final String ROLE_ADMIN = "admin";
    public static final String ROLE_MEMBER = "member";

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "room_id", nullable = false)
    private UUID roomId;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(nullable = false)
    private String role;

    @CreationTimestamp
    @Column(name = "joined_at", nullable = false, updatable = false)
    private Instant joinedAt;

    public boolean isAdmin() {
        return ROLE_ADMIN.equals(role);
    }
}
